libs/libdaemon/src/BaseDaemon.cpp

```diff
@@ -501,6 +501,26 @@ void BaseDaemon::closeFDs()
     }
 }
 
+namespace {
+/// In debug version on Linux, increase oom score so that clickhouse is killed
+/// first, instead of some service.
+#if defined(__linux__) and not defined(NDEBUG)
+void debug_adjust_oom_score()
+{
+    int file = open("/proc/self/oom_score_adj", O_WRONLY);
+    if (file < 0)
+    {
+        return;
```

Comment:
Throw an exception.

Reply:
I'd prefer the server to ignore the error and continue startup if something breaks here. After all, it's just a debugging aid. I'll log a warning if it fails.

Reply:
Supppose you have a mistake and write `om_score_adj` instead of `oom_score_adj`. Then we will never know about it. Or suppose Linux will change the path to this file or deprecate OOM killer at all. Or we don't have /proc filesystem in Docker...

PS. You can just write to log instead of throwing exception.